qutip/core/metrics.py

```diff
@@ -538,6 +522,9 @@ def dnorm(A, B=None, solver="CVXOPT", verbose=False, force_solve=False,
         # The 2-norm was not implemented for sparse matrices as of the time
         # of this writing. Thus, we must yet again go dense.
         return la.norm(op.full(), 2)
+    if not force_solve and J.iscptp:
```

Comment:
Let's do the `iscptp` check before the `iscp` one since it's a less general case and faster to calculate.